modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import ClayMultiSelect from '@clayui/multi-select';
+import classNames from 'classnames';
+import {openToast} from 'frontend-js-web';
+import React, {useContext, useState} from 'react';
+
+import ChartContext from '../ChartContext';
+import {createOrganizations} from '../data/organizations';
+
+export default function AddOrganizationModal({
+	closeModal,
+	observer,
+	parentData,
+}) {
+	const [query, setQuery] = useState('');
+	const [items, setItems] = useState([]);
+	const [errors, setErrors] = useState([]);
+	const {chartInstanceRef} = useContext(ChartContext);
+
+	function handleSave() {
+		const newOrganizations = query
+			? items.concat({label: query, value: query})
+			: [...items];
+		const organizationNames = newOrganizations.map((item) => item.label);
+
+		if (!newOrganizations.length) {
+			setErrors([
+				Liferay.Language.get('organization-name-field-required'),
+			]);
+
+			return;
+		}
+
+		createOrganizations(organizationNames, parentData).then((results) => {
+			const newOrganizationsDetails = [];
+			const newErrors = new Set();
+			const failedOrganizations = [];
+
+			results.forEach((result, fetchNumber) => {
+				if (result.status === 'rejected') {
+					failedOrganizations.push(newOrganizations[fetchNumber]);
+					newErrors.add(result.reason.title);
+				}
+				else {
+					newOrganizationsDetails.push(result.value);
+				}
+			});
+
+			setItems(failedOrganizations);
+			setErrors(Array.from(newErrors));
+			setQuery('');
```

Comment:
I think that we want empty lines between lines of code that execute stuff (anything except variable declarations)

Reply:
I would keep it this way for now. Will be changed in a next PR when this code will be updated anyway not to generate conflicts